tune_sklearn/utils.py

```diff
@@ -101,6 +103,14 @@ def get_early_stop_type(estimator, early_stopping):
 
     if not early_stopping:
         return EarlyStopping.NO_EARLY_STOP
+
+    if check_is_pipeline(estimator):
+        for step_name, step in estimator.steps:
```

Comment:
Two things:
- we shouldn't be only checking for regressors
- this is not foolproof & goes against duck typing principle of sklearn
- we can just assume that the last step of the pipeline is an estimator. You can't really make a pipeline with multiple estimators.